python/ray/train/huggingface/accelerate/accelerate_trainer.py

```diff
@@ -0,0 +1,422 @@
+                    )
+
+            torch.manual_seed(42)
+            train_dataset = ray.data.from_items(
+                [{"x": x, "y": 2 * x + 1} for x in range(200)]
+            )
+
+            # Define scaling and run configs
+            scaling_config = ScalingConfig(num_workers=3, use_gpu=use_gpu)
+            run_config = RunConfig(checkpoint_config=CheckpointConfig(num_to_keep=1))
+
+            trainer = AccelerateTrainer(
+                train_loop_per_worker=train_loop_per_worker,
+                # Instead of using a dict, you can run ``accelerate config``.
+                # The default value of None will then load that configuration
+                # file.
+                accelerate_config={},
+                scaling_config=scaling_config,
+                run_config=run_config,
+                datasets={"train": train_dataset},
+            )
+
+            result = trainer.fit()
+
+            best_checkpoint_loss = result.metrics["loss"]
+
+            # Assert loss is less 0.09
+            assert best_checkpoint_loss <= 0.09
+
+    .. testoutput::
+        :hide:
+        :options: +ELLIPSIS
+
+        ...
+
+    Args:
+        train_loop_per_worker: The training function to execute.
+            This can either take in no arguments or a ``config`` dict.
+        train_loop_config: Configurations to pass into
+            ``train_loop_per_worker`` if it accepts an argument.
+        accelerate_config: Accelerate configuration to be applied on every worker.
+            This can be a path to a file generated with ``accelerate config``,
+            a configuration dict or None, in which case it will load the configuration
+            file from the default location as defined by Accelerate.
+        torch_config: Configuration for setting up the PyTorch backend. If set to
+            None, use the default configuration. This replaces the ``backend_config``
+            arg of ``DataParallelTrainer``.
+        scaling_config: Configuration for how to scale data parallel training.
+        dataset_config: Configuration for dataset ingest.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Use
+            the key "train" to denote which dataset is the training
+            dataset. If a ``preprocessor`` is provided and has not already been fit,
+            it will be fit on the training dataset. All datasets will be transformed
+            by the ``preprocessor`` if one is provided.
+        preprocessor: A ``ray.data.Preprocessor`` to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        train_loop_per_worker: Union[Callable[[], None], Callable[[Dict], None]],
+        *,
+        train_loop_config: Optional[Dict] = None,
+        accelerate_config: Optional[Union[dict, str, Path, os.PathLike]] = None,
```

Comment:
this looks a bit weird, why is the new config param randomly placed at the 2nd place?
move it to the top?

Reply:
to me it felt it was nice to have the `train_loop_per_worker` and `train_loop_config` together, like in other Trainers - those are all keyword only arguments regardless (you can't pass them positionally)